# Report about bam file from GMI

I comfirmed whether tools work or not by SRR9974606. This files were generated by pipeline.sh. I used JGI5.1 genome as reference.
Used environment is docker image build from chip-env/Dockerfile.


## Bam files from GMI do not have BAM format

First, I executed `samtools sort` and `samtools index` commands for bam files by below script.

```bash
#!/bin/bash

# samtools sort and index
```bash
for file in `ls *.bam`; do
    echo ${file%.bam}
    sorted_bam=${file%.bam}_sorted.bam
    samtools sort $file > $sorted_bam
    samtools index $sorted_bam
done
```

Next, I executed `bamCoverage` to convert bam to bigwig format.

```bash
#!/bin/bash

# bam2bigwig

for file in `ls *_sorted.bam`; do
    bamCoverage --bam -o ${$file%.bam}.bw -of --bigwig 
done;
```

Below error occurred, so maybe bam files are not correct.

```bash
The file '105607_GCCAAT_HWMLYBGXC_1_20200114B_20200114_sorted.bam' does not have BAM or CRAM format
```

`bamCoverage` commands worked for SRR9974606 and sorted bam and bigwig were visualized by IGV(fig. 1).

---

<figure>
    <img src="2020-02-01-00-28-17.png", width=, height=>
    <figcaption>
        <b>Fig.1 Image of sorted bam and bigwig visualized by IGV</b><br>
    </figcaption>
</figure>

---

To confirm bam format, I executed `samtools view` command.

```bash
for file in `ls *.bam`; do 
    samtools view $file | head -n 100 > $file.n100.txt
done
```

As a result, bam files from GMI does not have information about location, mapping score etc., (./log/\*.bam.n100.txt). SRR9974606 bam is correct bam format and has information about location, mapping etc., (./log/ordinal_bam_n100.txt).

## Requests

If possible, I would like GMI to sent fastq files. Using gz compression, maybe size can be twice as big as bam files.  
if not possible, please sent correct bam fils.
